source/gui/addonGui.py

```diff
@@ -206,18 +221,26 @@ def refreshAddonsList(self,activeIndex=0):
 			self.helpButton.Disable()
 			self.removeButton.Disable()
 
+	def _shouldDisable(self, addon):
+		return not (addon.isPendingDisable or (addon.isDisabled and not addon.isPendingEnable and not addon.isPendingDisable))
+
 	def onListItemSelected(self, evt):
 		index=evt.GetIndex()
 		addon=self.curAddons[index] if index>=0 else None
+		# #3090: Change toggle button label to indicate action to be taken if clicked.
+		if addon is not None:
+			# Translators: The label for a button in Add-ons Manager dialog to enable or disable the selected add-on.
+			self.toggleButton.SetLabel(_("Enable add-on") if not self._shouldDisable(addon) else _("Disable add-on"))
 		self.aboutButton.Enable(addon is not None and not addon.isPendingRemove)
 		self.helpButton.Enable(bool(addon is not None and not addon.isPendingRemove and addon.getDocFilePath()))
+		self.toggleButton.Enable(addon is not None and not addon.isPendingRemove)
 		self.removeButton.Enable(addon is not None and not addon.isPendingRemove)
 
 	def onClose(self,evt):
 		self.Destroy()
 		if self.needsRestart:
-			# Translators: A message asking the user if they wish to restart NVDA as addons have been added or removed. 
-			if gui.messageBox(_("Add-ons have been added or removed. You must restart NVDA for these changes to take effect. Would you like to restart now?"),
+			# Translators: A message asking the user if they wish to restart NVDA as addons have been added, enabled/disabled or removed. 
+			if gui.messageBox(_("Add-ons have been added, enabled, disabled or removed. You must restart NVDA for these changes to take effect. Would you like to restart now?"),

```

Comment:
I see. The original intent was to tell users what can be done like what we have now (install/remove). But I can see this could be very complicated, especially when translating. I'll apply this and other UI suggestions. Thanks for catching this.